### PR TITLE
ESQL: Generate docs for the trig functions

### PR DESCRIPTION
**Pull request by nik9000**
This updates the in-code docs on the trig functions to line up with the docs, removes the docs, and uses the now mostly identical generated docs. This means we only need to document these functions in one place - right next to the code.

### CODE REVIEW

**Comment by leemthompo** [comment]
@nik9000 small point on phrasing:

I think "_returns the {foo} function of n_" or "_returns the {foo} function of a number_"

is clearer than simply "_returns the {foo} function_"

**Comment by nik9000** [comment]
> I think "_returns the {foo} function of n_" or "_returns the {foo} function of a number_"

I pushed 14e9c859462

File: docs/reference/esql/functions/types/cos.asciidoc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 [%header.monospaced.styled,format=dsv,separator=|]
 |===
-number | result
+angle | result
```
Comment (leemthompo) [inline docs/reference/esql/functions/types/cos.asciidoc:7]: is `angle` a valid type?

Comment (nik9000) [reply docs/reference/esql/functions/types/cos.asciidoc:7]: No, but that's the title row - it's the names of the columns.

**Review by leemthompo — APPROVED** [review]
Docs looking good, with that one note we chatted about— adding comments to files in `/parameters` to make it clear they're generated too :)

**Pull request merged**